### ISSUE TITLE
ReportThread - ModuleNotFoundError

### ISSUE THREAD

**Issue by trsavi** [jrieke/streamlit-analytics#14]
ReportThread seems to not be working with new version of streamlit (1.5.0.), I got ModuleNotFoundError. When I changed the source code in session_state.py and commented lines for importing report_thread and also disregarded ctx (ctx = ReportThread.get_report_ctx()) variable throughout the script I avoided error on local machine and everything worked fine (I could see basic statistics) but when I deploy it onto streamlit share framework error appeared once again. 
Any thoughts on this?

**Comment by chanansh** [comment]
please pull request the necessary changes so we can all enjoy the correction.

**Comment by chanansh** [comment]
@jrieke  please help correct this.

**Comment by riyadparvez** [comment]
An easy fix to this has already been posted.
https://discuss.streamlit.io/t/modulenotfounderror-no-module-named-streamlit-report-thread/20983/7?u=riyad_parvez

**Comment by yanirs** [comment]
This will be fixed by #16 or #12.

Until one of the PRs is merged, you can install one of the forks, e.g., `pip install git+https://github.com/yanirs/streamlit-analytics`.

**Comment by harishkashyap** [comment]
+1 on this. Same issue on sample code with latest python.

**Comment by alaturqua** [comment]
Any plans to fix this soon?

**Comment by corticalstack** [comment]
+1, same issue

**Comment by standard-outlier** [comment]
> This will be fixed by #16 or #12.
> 
> 
> 
> Until one of the PRs is merged, you can install one of the forks, e.g., `pip install git+https://github.com/yanirs/streamlit-analytics`.

Thanks I have tested you code, it works in streamlit 1.4, but getting another import error in 1.8

**Comment by franasal** [comment]
@yanirs I tried with your repo but unfortunately, it didn't work. I had to add these modifications to make it work for streamlit 1.8:

https://github.com/jrieke/streamlit-analytics/compare/main...franasal:streamlit-analytics:main
and added it to my `requirements.txt`:

`git+https://github.com/franasal/streamlit-analytics`

**Comment by yanirs** [comment]
Yeah, looks like this project isn't being maintained so I'm not using it. It'll probably break again with new Streamlit versions, unless it's actively maintained.

**Comment by fredzannarbor** [comment]
This should either be supported, or be available by default as part of all versions of Streamlit.

**Comment by jrieke** [comment]
Hey, sorry for the super long delay. I never had time to come around to this. I changed the implementation to work with `st.session_state` now, so it should be a lot more stable. Check out release 0.4.1, it's tested with Streamlit 1.13.0. (Note that the minimum Streamlit version is now 0.84.0).
